### PR TITLE
Multi user scenario support

### PR DESCRIPTION
**Pull request by noracato**
Adds multi user scenario support to saved scenarios
Goes with quintel/etengine#1410

I'll update this description later ;)

@Charlottevm feel free to check this PR for inspiration for the scenario history!

EDIT: semaphore is failing on bundle install, the tests are otherwise running fine. I'll fix that tonight

### CODE REVIEW

File: config/locales/en_etm.yml [hunk 1]
```diff
@@ -331,6 +332,43 @@ en:
...
     created_with: 'Scenario created with the Energy Transition Model'
+    users:
+      manage: 'Manage access'
+      manage_description:
+        Here you can manage who has access to your scenario and in which manner.
+        Click the 'Invite' button to give someone access to your scenario,
+        or use the table below to manage users that are already coupled.<br /><br />
+        Please visit the
+        <a href="https://docs.energytransitionmodel.com/main/#12-saved-scenario-access-management-page" target=\"_blank\">
+        documentation</a> for more information and details on user access management, such as user roles.
```
Comment (noracato) [inline config/locales/en_etm.yml:343]: Update this now that it is at the bottom of the page @mabijkerk @Charlottevm @kaskranenburgQ

File: config/locales/nl_etm.yml [hunk 2]
```diff
@@ -316,23 +317,65 @@ nl:
...
     created_with: 'Scenario gemaakt met het Energietransitiemodel'
+    users:
+      manage: 'Beheer toegang'
+      manage_description:
+        Hier kun je beheren wie met welke rechten toegang heeft tot je scenario.
+        Klik op de 'Gebruiker uitnodigen' knop om iemand toegang te geven, of gebruik
+        de tabel om gekoppelde gebruikers te beheren.<br /><br />
+        Bekijk alsjeblieft de
+        <a href="https://docs.energytransitionmodel.com/main/#12-saved-scenario-access-management-page" target=\"_blank\">
+        documentatien</a> voor meer informatie en details over gebruikersbeheer, zoals gebruikersrollen.
```
Comment (noracato) [inline config/locales/nl_etm.yml:345]: Update this now that it is at the bottom of the page @mabijkerk @Charlottevm @kaskranenburgQ

File: config/locales/nl_etm.yml [hunk 3]
```diff
@@ -316,23 +317,65 @@ nl:
...
+        base: Voer een e-mailadres in.
+        user_email: Het ingevoerde e-mail adres is onjuist.
+        update: Er ging iets mis tijdens het veranderen van deze rol!
+        destroy: Er ging iets mis tijdens het verwijderen van de gebruiker van dit scenario!
+        general: |
+          Probeer het later alsjeblieft nog een keer. Neem contact met ons op
+          indien deze fout zich blijft voordoen.
+        duplicate: Deze gebruiker is al uitgenodigd.
+    group: 'Groep'
     share:
```
Comment (noracato) [inline config/locales/nl_etm.yml:379]: Check Dutch translations @mabijkerk @Charlottevm @kaskranenburgQ

File: app/models/saved_scenario.rb [hunk 4]
```diff
@@ -141,11 +163,65 @@ def update_with_api_params(params)
...
+
+  # Returns a collection of SavedScenarioUsers
+  def collaborators
+    saved_scenario_users.where(role_id: User::ROLES.key(:scenario_collaborator))
+  end
+
+  # Returns a collection of SavedScenarioUsers
+  def viewers
+    saved_scenario_users.where(role_id: User::ROLES.key(:scenario_viewer))
+  end
```
[the remark this answers is not in the document]
Comment (noracato) [reply app/models/saved_scenario.rb:187]: Let's solve this later

File: config/locales/en_etm.yml [hunk 5]
```diff
@@ -336,12 +336,12 @@ en:
       manage: 'Manage access'
       manage_description:
         Here you can manage who has access to your scenario and in which manner.
-        Click the 'Invite' button to give someone access to your scenario,
+        Click the 'Invite users' button to give someone access to your scenario,
```
Comment (noracato) [inline config/locales/en_etm.yml:339]: Maybe 'invite others'? My fear is that with saying 'user's, people may think they can only invite registered users. But they can also invite people that are new to the ETM

File: config/locales/nl_etm.yml [hunk 6]
```diff
@@ -342,8 +342,8 @@ nl:
         de tabel om gekoppelde gebruikers te beheren.<br /><br />
         Bekijk alsjeblieft de
         <a href="https://docs.energytransitionmodel.com/main/#12-saved-scenario-access-management-page" target=\"_blank\">
-        documentatien</a> voor meer informatie en details over gebruikersbeheer, zoals gebruikersrollen.
-      invite: '+ Mensen uitnodigen'
+        documentatie</a> voor meer informatie en details over gebruikersbeheer, zoals gebruikersrollen.
+      invite: '+ Gebruiker uitnodigen'
```
Comment (noracato) [inline config/locales/nl_etm.yml:346]: Same here

File: config/locales/en_etm.yml (hunk 5 above)
Comment (kaskranenburgQ) [reply config/locales/en_etm.yml:339]: I agree, 'others' would be better

**Review by Charlottevm — APPROVED** [review]
LGTM 👍!

**Pull request merged**